python/cog/server/clients.py

```diff
@@ -188,17 +199,24 @@ async def upload_file(self, fh: io.IOBase, url: Optional[str]) -> str:
         # ensure trailing slash
         url_with_trailing_slash = url if url.endswith("/") else url + "/"
 
-        async def chunk_file_reader() -> AsyncIterator[bytes]:
-            while 1:
-                chunk = fh.read(1024 * 1024)
-                if isinstance(chunk, str):
-                    chunk = chunk.encode("utf-8")
-                if not chunk:
-                    log.info("finished reading file")
-                    break
-                yield chunk
+        class ChunkFileReader:
```

Comment:
What do you think about declaring `ChunkFileReader` as a top-level member?

Reply:
And passing in the file descriptor? No strong opinion, I'm happy to refactor it.